### PR TITLE
Add release notes for 0.266

### PR DESCRIPTION
**Pull request by varungajjala**
# Missing Release Notes
## James Petty
- [x] 1baba7e02f688d696307cf8732c2f2d71d8929c1 Add missing @Test annotation
- [x] 5c5152fe54860fbe2935330e06d8a657aa33c2a8 Fix AbstractArrayBlock#copyPositions with zero elements

# Extracted Release Notes
- #16923 (Author: Gurmeet Singh): upgrading the parquet/avro version
  - Parquet version upgraded from 1.10.1 (coming via presto-hive-apache 3.0.0-3) to 1.11.1 (coming via presto-hive-apache 3.0.0-7).
  - Avro version upgraded from 1.8.2 (coming via presto-hive-apache 3.0.0-3) to 1.9.2 (coming via presto-hive-apache 3.0.0-7).
- #16942 (Author: maobaolong): Support Caching module for iceberg connector
  - Support Caching module for iceberg connector.
- #16943 (Author: Xiang Fu): [Pinot] bugfixing: query failure and column ordering when doing Pinot SQL pushdown
  - Fix query crash when Pinot returns empty result.
  - Fix query crash when Pinot aggregation query returns mismatched schema.
- #16946 (Author: Beinan Wang): Upgrade Iceberg to 0.12.0 and upgrade Parquet to 1.12.0
  - Upgrade Parquet to 1.12.0.
  - Upgrade Iceberg to 0.12.0.
- #16949 (Author: Ajay George): Add skip checksum feature for verifier 
  - Add support to skip running checksum queries. This can be enabled by the configuration property ``skip-checksum``.
- #16954 (Author: Beinan Wang): Support creating iceberg v2 table
  - Support iceberg table with version1 and version2.
- #16956 (Author: Swapnil Tailor): Fix resource group concurrency for multi coordinator
  - Fix resource group concurrency for multi coordinator.
- #17005 (Author: Swapnil Tailor): Making resource group service cache configurable
  - Making resource group service cache configurable.

# All Commits
- 1baba7e02f688d696307cf8732c2f2d71d8929c1 Add missing @Test annotation (James Petty)
- 5c5152fe54860fbe2935330e06d8a657aa33c2a8 Fix AbstractArrayBlock#copyPositions with zero elements (James Petty)
- df317752eb74dd5a2cded874d3459c83007218ac Optimize validateMaterializedViewPartitionColumns function (penghuo)
- 9f81e053197ba185ad284c5cfcd68b14f353e118 Remove state from scalar operators (Sreeni Viswanadha)
- 522a6c8f939ad8e8167de52c4003563aa2975805 Wrap stream from HadoopInputFile to work around (panyliu)
- c5f8210f0c23953f3340edfe4a4c0fcb8183a391 Add tracer to session (tanjialiang)
- dd4dc7c9b261a87029dcce9a0f45c8e7ac294084 Making resource group service cache configurable (Swapnil Tailor)
- 1b229320a79270e510003030f4810e493361d767 Add skip checksum feature for verifier (Ajay George)
- 971eb7c7831b46f3a9712971afad6ee4ffdbe592 Fixing number of running queries (abhiseksaikia)
- b0cc508f32dc526f1c9353d3e1cf044cf410cb85 Invalidate Materialized view with subqueries (Pranjal Shankhdhar)
- ea1327ef469fe9b246a395f21f705a263d70ca4c Disable flaky test - testCleanBackupShards, testDeleteOldCompletedTransactions (Varun Gajjala)
- bc9ee2328569a1eb8a854db8f7fc785a4c738fc3 Add session property to force metadata query optimization (Zhan Yuan)
- 461c968fe4b5b655ff4104f46049a51e187ace04 Add tracer to session context (tanjialiang)
- 77978a42d65f14bbcb581ed09767eea1ca21e38c Improve the error message for IO related errors (Nikhil Collooru)
- 70f8f59e088974cd34de84b536260dcb06e52d11 Fix incorrect handling of System.nanoTime() in DriverContext (James Petty)
- 1d61a10ae2052e0fb238c9ce701373ec51a52536 Remove redundant traversal of OperatorStats (James Petty)
- 42f017a879ba2bb4008367abf3e96cd58b2295c3 Avoid intermediate collection creation in TaskStats creation (James Petty)
- 85fbbd403d3a61f10605a456dbc3a79c20a8d3d4 Optimize PipelineStats creation in PipelineContext (James Petty)
- 5fb1c8ac81f84e2c852d18f02f497f59eaeda001 Avoid redundant copies in TaskStats#summarize (James Petty)
- 54a56d7b65afccf409a509faf6c4896cd9afe7ef Avoid redundant copies in PipelineStats#summarize (James Petty)
- c821dc0d671f3d38aabd9ae2df2ac3604b26dedc Batch merge running operator stats in PipelineContext (James Petty)
- 937febafc7a91e7e52010323e888ca649984191f Opportunistically reuse instance in OperatorStats#summarize (James Petty)
- 2b266d68e46d4d238ad8bb3de4aacd367d0f9a56 Expose MapBlockBuilder KeyBlockBuilder and ValueBlockBuilder (Arunachalam Thirupathi)
- b12b848a17911e976e53e7c0c3eaa04b259112e7 Support iceberg table with version1 and version2 (Beinan Wang)
- 699832f664f02a9d1afb661d68d55dfb2a5ce3db Support Large Dictionary for OrcWriter (Arunachalam Thirupathi)
- fe936d01d67754654ec2a9f22a7a75919ff84e34 Optimize DictionaryCompressionOptimizer conversion code (Arunachalam Thirupathi)
- 426fbdd5d180d795d2a04731bf12f2685aec8f07 Fix the logic in LocalExchange for acquiring the bucket count (Ke Wang)
- 48fadf118502ac6d79aa56ec12754041e8b16ab5 Fix resource group concurrency for multi coordinator (Swapnil Tailor)
- 3fdbd185fe37a3d2368a2774595ef64552221c0b Fix batch read in SliceDirectSelectiveStreamReader (Masha Basmanova)
- 974b556fcf7c71f751f4e04adc20199e5edfa613 Fix failing to load hadoop filesystem of iceberg connector (panyliu)
- 3dfb5e044474712e561ea148453c2bd3bf0ee5cd Fix incorrect stats.numNulls value in Parquet (zhaoyulong)
- 3e030b5296631810a46b61197ed1229317d9f2e8 Fix worker-protocol documentation (Deepak Majeti)
- 3532c042756b23806ff7785f6f80d4f31aea868e Upgrade Iceberg to 0.12.0 and upgrade Parquet to 1.12.0 (Beinan Wang)
- dfebf44d9d95b30a971e99bd6fda22ecd452a00c Skip query failure with OOM if memory revoke is pending (Arjun Gupta)
- b7f31ded9cdd68ff6486941fc7e0a4de260c3113 Ignore Hadoop marker files when reading from S3. (sambekar15)
- d7ecb91915af0b534dc0f0c0177d6ab7898d83fc Support Caching module for iceberg connector (maobaolong)
- 9a1d36ab8ba81c12a601fb8af5598a8e2d2835f8 fixing column ordering for pinot sql syntax pushdown (Xiang Fu)
- 547a06813204b937635865452685c1855e02d2ed Revert "Add source location info to RowExpression" (Rongrong Zhong)
- b6b62349ee47677e1b8bc4d615dc8894a0ade586 upgrading parquet/avro to 1.11.1/1.9.2 respectively via (Gurmeet Singh)
- caf6fa8ca786fc3bf2a9c1c13787359ef5c59ba7 Implement efficient null handling for ArrayBlocks (James Petty)
- 99da0bbb0710f15f01b7bb5a4169e983db4b8efa Detect no null positions in RowBlock.fromFieldBlocks (James Petty)
- 8412fe36f0f5a6017685d2e77dda59f6e2ccb199 Implement efficient null handling for RowBlocks (James Petty)

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-0.266.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
+=============
+Release 0.266
+=============
+
+**Details**
+===========
+
+General Changes
+_______________
+* Fix resource group concurrency for multi coordinator.
```
Comment (ajaygeorge) [inline presto-docs/src/main/sphinx/release/release-0.266.rst:10]: Little bit unclear as to what is the user impact here. 
Maybe reword it on the lines of something like
Fix resource group concurrency for multi coordinator to reduce queuing time/ to address improper queuing logic

**Review by ajaygeorge — APPROVED** [review]
LGTM

File: presto-docs/src/main/sphinx/release/release-0.266.rst (hunk 1 above)
Comment (swapsmagic) [reply presto-docs/src/main/sphinx/release/release-0.266.rst:10]: Let's use this instead:
`Fix resource group concurrency for multi coordinator to not allow more than configured queries to run.`

File: presto-docs/src/main/sphinx/release/release-0.266.rst [hunk 2]
```diff
@@ -0,0 +1,39 @@
+=============
+Release 0.266
+=============
+
+**Details**
+===========
+
+General Changes
+_______________
+* Fix resource group concurrency in multi coordinator to not allow more than configured queries to run.
```
Comment (mayankgarg1990) [inline presto-docs/src/main/sphinx/release/release-0.266.rst:10]: Is multi-coordinator out of beta - @tdcmeehan  ?

**Review by mayankgarg1990 — APPROVED** [review]

File: presto-docs/src/main/sphinx/release/release-0.266.rst [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+Pinot Changes
+_____________
+* Fix query crash when Pinot aggregation query returns mismatched schema.
+* Fix query crash when Pinot returns empty result.
+
+Verifier Changes
+________________
+* Add support to skip running checksum queries. This can be enabled with the configuration property ``skip-checksum``.
+
+Resource Group Changes
```
Comment (tdcmeehan) [inline presto-docs/src/main/sphinx/release/release-0.266.rst:32]: Let's remove this until we mark disaggregated coordinators as GA.

File: presto-docs/src/main/sphinx/release/release-0.266.rst [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+Iceberg Changes
+_______________
+* Add support for caching module in Iceberg connector.
+* Add support to create Iceberg tables with version1 and version2.
+* Upgrade Iceberg to 0.12.0.
+
+Pinot Changes
+_____________
+* Fix query crash when Pinot aggregation query returns mismatched schema.
+* Fix query crash when Pinot returns empty result.
```
Comment (tdcmeehan) [inline presto-docs/src/main/sphinx/release/release-0.266.rst:26]:
```suggestion
* Fix query error when Pinot aggregation query returns mismatched schema.
* Fix query error when Pinot returns empty result.
```

File: presto-docs/src/main/sphinx/release/release-0.266.rst [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+* Fix resource group concurrency in multi coordinator to not allow more than configured queries to run.
+
+Hive Changes
+____________
+* Upgrade Parquet to 1.12.0.
+* Upgrade Avro to 1.9.2.
+
+Iceberg Changes
+_______________
+* Add support for caching module in Iceberg connector.
```
Comment (tdcmeehan) [inline presto-docs/src/main/sphinx/release/release-0.266.rst:19]:
```suggestion
* Add support for caching in Iceberg connector.
```

File: presto-docs/src/main/sphinx/release/release-0.266.rst [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+* Add support to create Iceberg tables with version1 and version2.
```
Comment (tdcmeehan) [inline presto-docs/src/main/sphinx/release/release-0.266.rst:20]: Seems like weird wording--what does version1 and version2 mean?

File: presto-docs/src/main/sphinx/release/release-0.266.rst [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+* Upgrade Iceberg to 0.12.0.
```
Comment (tdcmeehan) [inline presto-docs/src/main/sphinx/release/release-0.266.rst:21]: Is this relevant?

**Review by highker — APPROVED** [review]
Looks good % @tdcmeehan's comment

File: presto-docs/src/main/sphinx/release/release-0.266.rst (hunk 2 above)
Comment (tdcmeehan) [reply presto-docs/src/main/sphinx/release/release-0.266.rst:10]: Let's remove this
